experimental/list_formatter/Cargo.toml

```diff
@@ -44,4 +44,4 @@ std = ["icu_provider/std", "icu_locid/std"]
 default = ["provider_serde"]
 bench = []
 provider_serde = ["serde", "zerovec/serde", "deduplicating_array"]
-provider_transform_internals = ["std"]
```

Comment:
Question: why did you get rid of the other feature `provider_transform_internals`?

Reply:
DCE covers it, because it was only used to hide functions, and I think it's good to reduce feature complexity.

Reply:
I would like to maintain a better separation between what code is needed at runtime versus for the data building. If we don't use a feature, we should at least move the code somehow into another module. There is prior art for this in the PluralRules crate, which uses "runtime" and "reference".

Reply:
Thoughts from @Manishearth:

- Removing code with a feature improves build times, and helps when building code for odd platforms like Cortex/WASM
- But I'm not happy with the explosion of features we're having; should we consider consolidating the features? A new feature called "datagen" that includes provider_transform_internals + serialize?
- In the short term, don't get rid of the feature, but open an issue to consolidate the features.

Reply:
There's #1320